api/client_test.go

```diff
@@ -0,0 +1,274 @@
+// Copyright (c) 2019-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package api
+
+import (
+	"fmt"
+	"net/http/httptest"
+	"sync"
+	"testing"
+	"time"
+
+	agentClient "github.com/mattermost/mattermost-load-test-ng/api/client/agent"
+	coordClient "github.com/mattermost/mattermost-load-test-ng/api/client/coordinator"
+	"github.com/mattermost/mattermost-load-test-ng/coordinator"
+	"github.com/mattermost/mattermost-load-test-ng/defaults"
+	"github.com/mattermost/mattermost-load-test-ng/loadtest"
+	"github.com/mattermost/mattermost-load-test-ng/loadtest/control/simulcontroller"
+	"github.com/mattermost/mattermost-load-test-ng/logger"
+
+	"github.com/stretchr/testify/require"
+)
+
+// concurrency level
+const n = 8
+
+func TestAgentClientConcurrency(t *testing.T) {
+	// create http.Handler
+	handler := SetupAPIRouter(logger.New(&logger.Settings{}), logger.New(&logger.Settings{}))
+
+	// run server using httptest
+	server := httptest.NewServer(handler)
+	defer server.Close()
+
+	var wg sync.WaitGroup
+
+	t.Run("create", func(t *testing.T) {
+		wg.Add(n)
+		id := "test"
+		var success int
+		var ltConfig loadtest.Config
+		var ucConfig simulcontroller.Config
+		defaults.Set(&ltConfig)
+		defaults.Set(&ucConfig)
+		agent, err := agentClient.New(id, server.URL, nil)
+		require.NoError(t, err)
+		require.NotNil(t, agent)
+		for i := 0; i < n; i++ {
+			go func() {
+				if _, err := agent.Create(&ltConfig, &ucConfig); err == nil {
+					success += 1
```

Comment:
Are we relying on the correctness of `agent.Create` that multiple goroutines won't write to the success variable?

Reply:
Yes. I could have used atomics but thought that the race would happen only if the test failed so I kept it simple.
We can add a comment if it looks too sketchy :P

Reply:
Comment would be good. :+1: